tests/parsing/FilterHelperTest.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+
+from coalib.parsing.FilterHelper import (
+    apply_filter,
+    apply_filters,
+    is_valid_filter,
+    _filter_section_bears
+)
+from coalib.parsing.filters import available_filters
+from coalib.parsing.InvalidFilterException import InvalidFilterException
+from coalib.settings.ConfigurationGathering import get_all_bears
+
+
+class FilterHelperTest(unittest.TestCase):
+
+    def test_apply_filter_exception(self):
+        with self.assertRaises(InvalidFilterException) as exp:
+            apply_filter('unknown', ['args'])
+
+        message = str(exp.exception)
+        self.assertEqual("'unknown' is an invalid filter. Available filters: "
+                         + ', '.join(sorted(available_filters)), message)
+
+    def test_apply_filters_exception(self):
+        with self.assertRaises(InvalidFilterException) as exp:
+            apply_filters([['language', 'C', 'Python'],
```

Comment:
what is this?

you are passing an array of arrays into the arg `filter_name` arg... intentional?

Reply:
the exception is the same as above, so it doesnt appear that this test case provides anything more than the above

Reply:
This case is for `apply_filters(...)` and above one is for `apply_filter(...)`.